### PR TITLE
Task/validation connect front

### PR DESCRIPTION
**Pull request by SkrypnikMR**

### CODE REVIEW

File: client/src/Store/login/sagas.js [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
 export function* workerLogin() {
     try {
         const data = yield select(logValues);
-        const serverAnswer = yield call(postRequest, routes.account.login, data);
-        if (serverAnswer.token) {
+        const { loginValidation } = validation;
+        const { message: validateMessage, isValid } = loginValidation(data);
+        if (!isValid) {
+            return NotificationManager
+                .error(`${i18next.t(validateMessage.replace(/ /g, '_').toLowerCase())}`, `${i18next.t('input_error')}`, 2000);
```
Comment (MaxShvachko) [inline client/src/Store/login/sagas.js]: rework notification messages and delete this logic

File: client/src/Store/login/sagas.js [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
+        const { token, message } = yield call(postRequest, routes.account.login, data);
+        if (token) {
             yield (put(clearLoginInputs()));
             yield put(setLoginValue({ name: 'success', value: true }));
             yield put(reciveSuccessRequest());
         } else {
             yield put(setLoginValue({ name: 'success', value: false }));
             yield put(reciveErrorRequest());
+            return NotificationManager
+                .error(`${i18next.t(message.replace(/ /g, '_').toLowerCase())}`, `${i18next.t('login_error')}`, 2000);
```
Comment (MaxShvachko) [inline client/src/Store/login/sagas.js]: the same

File: client/src/Store/login/sagas.js [hunk 3]
```diff
@@ -1,25 +1,38 @@
...
         }
     } catch (e) {
         yield put(setLoginValue({ name: 'success', value: false }));
         yield put(reciveErrorRequest());
+        return NotificationManager
+            .error(`${i18next.t('server_error_text')}`, `${i18next.t('server_error')}`, 2000);
```
Comment (MaxShvachko) [inline client/src/Store/login/sagas.js]: if you can, delete quots
and pass only 

` .error(i18next.t('server_error_text'), i18next.t('server_error'), 2000);`

File: client/src/Store/registration/sagas.js [hunk 4]
```diff
@@ -1,25 +1,38 @@
...
 import { postRequest } from '/src/helpers/requests';
-import { routes } from '/src/contsants/routes';
+import { routes } from '/src/constants/routes';
+import { NotificationManager } from 'react-notifications';
+import i18next from 'i18next';
 import { actionTypes } from './actionTypes';
 import { regValues } from './selectors';
 import { setRegistrationValue, clearRegistrationInputs, reciveErrorRequest, reciveSuccessRequest } from './actions';
+import { validation } from '../../helpers/validation';
+
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js]: empty line

File: client/src/Store/registration/sagas.js [hunk 5]
```diff
@@ -1,25 +1,38 @@
 import { takeEvery, call, select, put } from 'redux-saga/effects';
 import { postRequest } from '/src/helpers/requests';
-import { routes } from '/src/contsants/routes';
+import { routes } from '/src/constants/routes';
+import { NotificationManager } from 'react-notifications';
+import i18next from 'i18next';
 import { actionTypes } from './actionTypes';
 import { regValues } from './selectors';
 import { setRegistrationValue, clearRegistrationInputs, reciveErrorRequest, reciveSuccessRequest } from './actions';
+import { validation } from '../../helpers/validation';
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js]: maybe use path 'src/helpers/validation'

File: client/src/Store/registration/sagas.js [hunk 6]
```diff
@@ -1,25 +1,38 @@
...
 export function* workerRegistration() {
     try {
         const data = yield select(regValues);
-        const serverAnswer = yield call(postRequest, routes.account.registration, data);
-        if (serverAnswer.message === 'done') {
+        const { registrationValidation } = validation;
+        const { message: validateMessage, isValid } = registrationValidation(data);
+        if (!isValid) {
+            return NotificationManager
+                .error(`${i18next.t(validateMessage.replace(/ /g, '_').toLowerCase())}`, `${i18next.t('input_error')}`, 2000);
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js]: the same

File: client/src/Store/registration/sagas.js [hunk 7]
```diff
@@ -1,25 +1,38 @@
...
+        }
+        const { message } = yield call(postRequest, routes.account.registration, { ...data, confirm: undefined });
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js:19]: if you can't get object from response and destructure message your saga will crash

File: client/src/Store/registration/sagas.js [hunk 8]
```diff
@@ -1,25 +1,38 @@
...
+        if (message === 'done') {
             yield (put(clearRegistrationInputs()));
+
             yield put(setRegistrationValue({ name: 'success', value: true }));
             yield put(reciveSuccessRequest());
         } else {
             yield put(setRegistrationValue({ name: 'success', value: false }));
             yield put(reciveErrorRequest());
+            return NotificationManager
+                .error(`${i18next.t(message.replace(/ /g, '_'))}`, `${i18next.t('reg_error')}`, 2000);
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js]: the same

File: client/src/Store/registration/sagas.js [hunk 9]
```diff
@@ -1,25 +1,38 @@
...
         }
     } catch (e) {
         yield put(setRegistrationValue({ name: 'success', value: false }));
-        yield put(reciveErrorRequest());
+        yield put(reciveErrorRequest(''));
```
Comment (MaxShvachko) [inline client/src/Store/registration/sagas.js]: For what ?

File: client/src/components/App/App.js [hunk 10]
```diff
@@ -1,11 +1,14 @@
...
 import PropTypes from 'prop-types';
+import { NotificationContainer } from 'react-notifications';
+import 'react-notifications/lib/notifications.css';
 import Registration from '../Registration';
 import Login from '../Login';
-import { APP_ROUTES } from '/src/contsants/reactRoutes';
+import { APP_ROUTES } from '/src/constants/reactRoutes';
 import { StAppDiv } from './styled';
 
+
```
Comment (MaxShvachko) [inline client/src/components/App/App.js]: empty line

File: client/src/components/Registration/Registration.jsx [hunk 11]
```diff
@@ -1,17 +1,23 @@
...
 import Button from '../UI/Button';
 
 const Registration = ({ sendRegistrationRequest, setRegistrationValue, fields }) => {
     const { t } = useTranslation();
-    if (fields.success) return <Redirect to={APP_ROUTES.login} />;
+    const { success } = fields;
+    if (success) return <Redirect to={APP_ROUTES.login} />;
+    const handleOnChange = (data) => {
+        const { error } = fields;
+        if (error) setRegistrationValue({ name: 'error', value: '' });
```
Comment (MaxShvachko) [inline client/src/components/Registration/Registration.jsx]: думаю null подойдет лучше, так как у тебя в initialState error: null

File: client/src/helpers/validation.js [hunk 12]
```diff
@@ -4,9 +4,9 @@ export const validation = {
     registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
         if (!regExp.emailRegExp.test(email) || !email) return { message: 'Invalid email', isValid: false };
         if (!regExp.passwordRegExp.test(password) || !password) return { message: 'Invalid password', isValid: false };
-        if (!regExp.firstAndLastNameRegExp.test(firstName) || !firstName) return { message: 'Invalid firstName', isValid: false };
-        if (!regExp.firstAndLastNameRegExp.test(lastName) || !lastName) return { message: 'Invalid lastName', isValid: false };
         if (password !== confirm) return { message: 'Password mismatch', isValid: false };
+        if (!regExp.firstAndLastNameRegExp.test(firstName) || !firstName) return { message: 'Invalid first Name', isValid: false };
+        if (!regExp.firstAndLastNameRegExp.test(lastName) || !lastName) return { message: 'Invalid last Name', isValid: false };
```
Comment (MaxShvachko) [inline client/src/helpers/validation.js]: write message in snake_case

File: client/src/i18n.js [hunk 13]
```diff
@@ -5,7 +5,7 @@ import en from './locales/en.json';
 import ar from './locales/ar.json';
 import ru from './locales/ru.json';
 
-const language = localStorage.getItem('lang') || 'en';
+const language = localStorage.getItem('lang') || 'ar';
```
Comment (MaxShvachko) [inline client/src/i18n.js]: in default Arabian? ))

Comment (SkrypnikMR) [reply client/src/i18n.js]: ))) my bad, but it would be interesting

File: client/src/Store/registration/sagas.js (hunk 7 above)
Comment (SkrypnikMR) [reply client/src/Store/registration/sagas.js:19]: I always get message, on 200, 400, 500, status, if server not work - we faling in catch block with 
"server shutdown" message to user, don't see problem here.

File: client/src/Store/registration/sagas.js (hunk 9 above)
Comment (SkrypnikMR) [reply client/src/Store/registration/sagas.js]: my bad

File: client/src/Store/registration/sagas.js (hunk 5 above)
Comment (SkrypnikMR) [reply client/src/Store/registration/sagas.js]: 2 ../ it's bad?

Comment (SkrypnikMR) [reply client/src/Store/registration/sagas.js]: i did it, but think 2 it's not bad)

File: client/src/components/Registration/Registration.jsx (hunk 11 above)
Comment (SkrypnikMR) [reply client/src/components/Registration/Registration.jsx]: Пустая же строка в проверке сработает так же как нал )) Но я исправил

File: client/src/helpers/validation.js (hunk 12 above)
Comment (SkrypnikMR) [reply client/src/helpers/validation.js]: Написал всё снейком и ответы с сервера и все валидации и переделал в тестах

**Pull request merged**